### PR TITLE
New Component to manager Legends

### PR DESCRIPTION
**Pull request by cassunde**
**Task**

#15 

**Goal**

Refactore Bar Component, moving all code about legend to Legend component

### CODE REVIEW

File: src/js/components/Legenda/index.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        label: 'GP',
+        legenda: 'Gabinete do Prefeito',
+    },
+    {
+        label: 'GVP',
+        legenda: 'Gabinete do Vice Prefeito',
+    },
+];
+
+export function encontraLegenda(label) {
```
Comment (samuelsilvadev) [inline src/js/components/Legenda/index.js]: I think if we want to use this function as a Component we should use capital letter here and pass the label as prop


```js
export default function Legenda(label) {
 const legendaEncontrada = legendas.find(legenda => legenda.label === label);
 return legendaEncontrada && legendaEncontrada.legenda;
}
```

and use it like

```js
<Legend label={'something'} 
```

what do u think?

Comment (cassunde) [reply src/js/components/Legenda/index.js]: How to use this component in the Graphics Bar component?
```js
<td><Legend label={'something'} /> </td>
```

Comment (samuelsilvadev) [reply src/js/components/Legenda/index.js]: yeap, I think it's great :D

File: src/js/components/Legenda/index.js [hunk 2]
```diff
@@ -33,9 +37,53 @@ const legendas = [
...
-
+function encontraLegenda(label) {
     const legendaEncontrada = legendas.find(legenda => legenda.label === label);
 
     return legendaEncontrada && legendaEncontrada.legenda;
-}
\ No newline at end of file
+}
+
+export default class Legenda extends Component {
```
Comment (samuelsilvadev) [inline src/js/components/Legenda/index.js]: I think this component can be just stateless

File: src/js/components/Legenda/index.js [hunk 3]
```diff
@@ -33,9 +37,53 @@ const legendas = [
...
+    constructor(props) {
+        super(props);
+
+        this.state = {
```
Comment (samuelsilvadev) [inline src/js/components/Legenda/index.js]: we don't need state in this component, I guess just the prop `labels` it's enough.

File: src/js/components/Legenda/index.js [hunk 4]
```diff
@@ -33,9 +37,53 @@ const legendas = [
...
+        return (
+            <Table>
+                <thead>
+                    <tr>
+                        <th>Sigla</th>
+                        <th>Legenda</th>
+                    </tr>
+                </thead>
+                <tbody>
+                    {labels.map((label, index) => {
```
Comment (samuelsilvadev) [inline src/js/components/Legenda/index.js]: here we would use the prop `labels` and not the state `labels`

Comment (samuelsilvadev) [inline src/js/components/Legenda/index.js]: can we create a separated render function?
passing the callback directly yo map we are creating a new function in each iteration

File: src/js/components/Barra/index.js [hunk 5]
```diff
@@ -1,10 +1,10 @@
 import React, { Component } from 'react';
 import PropTypes from 'prop-types';
-import { Row, Col, Table } from 'reactstrap';
+import { Row, Col } from 'reactstrap';
 import { Bar } from 'react-chartjs-2';
 import axios from 'axios';
 
-import * as Legenda from '../Legenda'
+import Legenda from '../Legenda';
```
Comment (samuelsilvadev) [inline src/js/components/Barra/index.js]: can we put this import right below `import Cartao from '../Cartao';`? just for we keep our components together

**Comment by cassunde** [comment]
Maybe this component can be irrelevant, because this information can be displayed using this function

```js
function getFormattedLabel(tooltipItem, data)
```

What you think ?

**Comment by samuelsilvadev** [comment]
I see.. humm..., I liked to see the legend below the chart, but I agree that it's useless since the user can see on the chart directly. 

about the space I think the page always can grow so it's not a problem.

**Comment by samuelsilvadev** [comment]
if you think it's better without this component I agree too, it was just a personal taste for the matter :D

**Comment by cassunde** [comment]
Ok, let's create a new problem to include the caption in the chart, today I update this pr with changes

**Comment by cassunde** [comment]
close this pr because the pr #20 best represents the solution to issue #16

**Pull request closed without merging**